src/Group.jl

```diff
@@ -0,0 +1,288 @@
+end
+
+"""
+    exp!(G::AbstractGroupManifold, y, v, t::Real)
+
+Exponential map of tangent vector `t*v` at the identity element of group `G`.
+Result is saved to `y`.
+"""
+exp!(G::AbstractGroupManifold, y, v, t::Real) = exp!(G, y, t*v)
+
+function exp!(G::AbstractGroupManifold, y, x, v)
+    exp!(G, y, \(G, x, v))
+    y .= compose(G, x, y)
+    return y
+end
+
+"""
+    exp(G::AbstractGroupManifold, v)
+
+Exponential map of tangent vector `v` at the identity element of group `G`.
+"""
+function exp(G::AbstractGroupManifold, v)
+    y = similar(v)
+    exp!(G, y, v)
+    return y
+end
+
+"""
+    exp(G::AbstractGroupManifold, v, t::Real)
+
+Exponential map of tangent vector `t*v` at the identity element of group `G`.
+"""
+exp(G::AbstractGroupManifold, v, t::Real) = exp(G, t*v)
+
+function log!(G::AbstractGroupManifold, v, y)
+    error("Logarithmic map not implemented on $(typeof(G)) for point $(typeof(y)) and vector $(typeof(v))")
+end
+
+function log!(G::AbstractGroupManifold, v, x, y)
+    log!(G, v, \(G, x, y))
+    v .= left_action(G, x, v)
+end
+
+function log(G::AbstractGroupManifold, y)
+    v = similar_result(G, log, y)
+    log!(G, v, y)
+    return v
+end
+
+
+"""
+    AdditionOperation <: AbstractGroupOperation
+
+Group operation that consists of simple addition.
+"""
+struct AdditionOperation <: AbstractGroupOperation end
+
+inv(::AbstractGroupManifold{AdditionOperation}, x) = -x
+
+identity(::AbstractGroupManifold{AdditionOperation}, x) = zero(x)
+
+left_action(::AbstractGroupManifold{AdditionOperation}, x, p) = x + p
+
+@doc doc"""
+    right_action(::AbstractGroupManifold{AdditionOperation}, x, p)
+
+For an element $p$ of some set, compute the right action of group element $x$
+on $p$, i.e. $p \cdot x$. For an `AdditionOperation`, the right and left
+actions are the same.
+"""
+function right_action(G::AbstractGroupManifold{AdditionOperation}, x, p)
+    return left_action(G, x, p)
+end
+
+compose(::AbstractGroupManifold{AdditionOperation}, x...) = +(x...)
+
+lie_bracket(::AbstractGroupManifold{AdditionOperation}, v, w) = zero(v)
+
+exp!(::AbstractGroupManifold{AdditionOperation}, y, v) = y .= v
+
+log!(::AbstractGroupManifold{AdditionOperation}, v, y) = v .= y
+
+
+"""
+    MultiplicationOperation <: AbstractGroupOperation
+
+Group operation that consists of multiplication.
+"""
+struct MultiplicationOperation <: AbstractGroupOperation end
+
+inv(::AbstractGroupManifold{MultiplicationOperation}, x) = inv(x)
+
+identity(::AbstractGroupManifold{MultiplicationOperation}, x) = one(x)
+
+left_action(::AbstractGroupManifold{MultiplicationOperation}, x, p) = x * p
+
+compose(::AbstractGroupManifold{MultiplicationOperation}, x...) = *(x...)
+
+/(::AbstractGroupManifold{MultiplicationOperation}, x, y) = x / y
+
+\(::AbstractGroupManifold{MultiplicationOperation}, x, y) = x \ y
+
+@traitfn function lie_bracket(G::GT, v, w) where {O<:MultiplicationOperation,
```

Comment:
Wouldn't it make sense to add here another argument: the point `x` at which `v` and `w` are tangent?

Reply:
Yes, that's a good point

Reply:
Added in a version of `lie_bracket` for left-invariant fields